prefect_utils/snowflake.py

```diff
@@ -69,21 +71,272 @@ def qualified_stage_name(database, schema, table) -> str:
     )
 
 
+def qualified_pipe_name(database, schema, table) -> str:
+    """
+    Fully qualified Snowpipe name.
+    """
+    return "{database}.{schema}.{table}_pipe".format(
+        database=database, schema=schema, table=table,
+    )
+
+
 @task
 @backoff.on_exception(backoff.expo,
                       snowflake.connector.ProgrammingError,
                       max_tries=3)
-def load_ga_data_to_snowflake(
+def load_s3_data_to_snowflake_pipe(
+    sf_credentials: dict,
+    sf_database: str,
+    sf_schema: str,
+    sf_table: str,
+    sf_role: str,
+    sf_warehouse: str,
+    sf_storage_integration: str,
+    s3_url: str,
+    sf_file_format_type: str = 'JSON'
+):
+    """
+    Loads from S3 objects to Snowflake using a Snowpipe.
+    Args:
+      sf_credentials (dict):
+        Snowflake public key credentials in the format required by create_snowflake_connection.
+      sf_database (str): Name of the destination database.
+      sf_schema (str): Name of the destination schema.
+      sf_table (str): Name of the destination table.
+      sf_role (str): Name of the snowflake role to assume.
+      sf_warehouse (str): Name of the Snowflake warehouse to be used for loading.
+      sf_storage_integration (str): The name of the pre-configured storage integration created for this flow.
+      s3_url (str): Full URL to the S3 path containing the files to load.
+      sf_file_format_type (str, optional): Snowflake file format for the Stage. Defaults to 'JSON'.
+    """
+    sf_connection = create_snowflake_connection(sf_credentials, sf_role, warehouse=sf_warehouse)
+
+    try:
+        # Create the generic loading table
+        query = """
+        CREATE TABLE IF NOT EXISTS {table} (
+            ID NUMBER AUTOINCREMENT START 1 INCREMENT 1,
+            LOAD TIMESTAMP_LTZ DEFAULT CURRENT_TIMESTAMP(),
+            ORIGIN_FILE_NAME VARCHAR(16777216),
+            ORIGIN_FILE_LINE NUMBER(38,0),
+            ORIGIN_STR VARCHAR(16777216),
+            PROPERTIES VARIANT
+        );
+        """.format(
+            table=qualified_table_name(sf_database, sf_schema, sf_table)
+        )
+
+        sf_connection.cursor().execute(query)
+
+        # Create the loading stage
+        query = """
+        CREATE OR REPLACE STAGE {stage_name}
```

Comment:
A small detail, but is this going to work, or does this need to be CREATE STAGE IF NOT EXISTS?   Not sure if it will replace the stage if it is already in use by a PIPE.  I could imagine it going either way....

Reply:
This was going to be one of my questions for the folks who worked on BQ (from where it was copied), but I believe that the reason it's like this because it self-heals of the S3 URL changes: https://docs.snowflake.com/en/sql-reference/sql/alter-stage.html#external-stage-parameters-externalstageparams

Reply:
I didn't write this code, but I will say that I hadn't given that situation (changing URLs of stages) any thought.   I guess my biggest concern is whether creating and re-creating the stage every time will actually work, or if it will cause the pipe to re-load every file every time.  I had always thought the "memory" of which files were loaded was preserved in the stage, but I could be wrong.